transformer_engine/pytorch/csrc/extensions/multi_tensor/multi_tensor_adam.cu

```diff
@@ -171,6 +172,9 @@ struct AdamFunctorExtraOutDtype {
         if (fp8_data.amax_ptr != nullptr) {
           transformer_engine::atomicMaxFloat(fp8_data.amax_ptr, fp8_data.max);
         }
+        if (fp8_data.scale_inv_ptr != nullptr) {
+          transformer_engine::reciprocal(fp8_data.scale_inv_ptr, fp8_data.scale);
+        }
```

Comment:
> Hi @timmoon10 , I encountered an issue when trying to update scale_inv inside the Adam kernel using `*scale_inv_ptr = 1.0f / scale`. This resulted in loss not being bit-wise aligned. The reason is that TE/PyTorch compilation uses `--use_fast_math`, which compiles the reciprocal calculation into a single `MUFU.RCP` instruction, producing an approximate result rather than an accurate one. To achieve bit-wise alignment of the loss, I had to update scale_inv outside the Adam kernel. This also leads to suboptimal performance. Do you have any suggestions to address this?

Were you able to work around the `--use_fast_math` issue? One workaround is to replace `1 / scale` with [`__frcp_rn`](https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__INTRINSIC__SINGLE.html#group__CUDA__MATH__INTRINSIC__SINGLE_1gba455801af8ac9af405a5d37ef2f077b). Looking at [the programming guide](https://docs.nvidia.com/cuda/cuda-c-programming-guide/#intrinsic-functions), reciprocal is not affected by `--use_fast_math` while division is.

Reply:
Done. `__frcp_rn` can solve this issue.